utility/resource_loader_compat.h

```diff
@@ -0,0 +1,281 @@
+	uint32_t engine_ver_minor = 0;
+	bool stored_big_endian = false;
+	bool stored_use_real64 = false;
+	bool convert_v2image_indexed = false;
+	bool hacks_for_deprecated_v2img_formats = true;
+	bool no_abort_on_ext_load_fail = true;
+	FileAccess *f = nullptr;
+
+	uint64_t importmd_ofs = 0;
+
+	Vector<char> str_buf;
+	Vector<StringName> string_map;
+
+	StringName _get_string();
+
+	struct ExtResource {
+		String path;
+		String type;
+		RES cache;
+	};
+
+	bool use_sub_threads = false;
+	float *progress = nullptr;
+	Vector<ExtResource> external_resources;
+	struct PropPair {
+		StringName name;
+		Variant value;
+	};
+	struct IntResource {
+		String path;
+		uint64_t offset;
+	};
+
+
+	Vector<IntResource> internal_resources;
+	Map<String, RES> internal_index_cache;
+	Map<String, String> internal_type_cache;
+	Map<String, List<ResourceProperty>> internal_index_cached_properties;
+
+	ResourceFormatLoader::CacheMode cache_mode = ResourceFormatLoader::CACHE_MODE_IGNORE;
+	void save_unicode_string(const String &p_string);
+	static void save_ustring(FileAccess * f, const String &p_string);
+	Error repair_property(const String &rtype, StringName &name, Variant &value);
+
+	String get_unicode_string();
+	static void advance_padding(FileAccess * f, uint32_t p_len);
+	void _advance_padding(uint32_t p_len);
+
+	Map<String, String> remaps;
+	Error error = OK;
+
+	friend class ResourceFormatLoaderCompat;
+	friend class TextureLoaderCompat;
+	static Map<String,String> _get_file_info(FileAccess *f, Error *r_error);
+	Error load_import_metadata();
+	static Error _get_resource_header(FileAccess *f);
+	RES set_dummy_ext(const String& path, const String& exttype);
+	RES set_dummy_ext(const uint32_t erindex);
+	RES make_dummy(const String& path, const String& type, const uint32_t subidx);
+	void debug_print_properties(String res_name, String res_type, List<ResourceProperty> lrp);
+	
+	Error load_ext_resource(const uint32_t i);
+	RES get_external_resource(const int subindex);
+	RES get_external_resource(const String &path);
+	bool has_external_resource(const String &path);
+
+	RES instance_internal_resource(const String &path, const String &type, const int subindex);
+	RES get_internal_resource(const int subindex);
+	RES get_internal_resource(const String &path);
+	String get_internal_resource_type(const String &path);
+	bool has_internal_resource(const String &path);
+	List<ResourceProperty> get_internal_resource_properties(const String &path);
+
+	static String get_resource_path(const RES &res);
+	
+	Error load_internal_resource(const int i);
+	Error real_load_internal_resource(const int i);
+	Error open_text(FileAccess *p_f, bool p_skip_first_tag);
+	Error write_variant_bin(FileAccess *fa, const Variant &p_property, const PropertyInfo &p_hint = PropertyInfo());
+	Error parse_variant(Variant &r_v);
+	Map<String, RES> dependency_cache;
+    public:
+		void ResourceLoaderBinaryCompat::get_dependencies(FileAccess *p_f, List<String> *p_dependencies, bool p_add_types);
```

Comment:
```suggestion
		void get_dependencies(FileAccess *p_f, List<String> *p_dependencies, bool p_add_types);
```
Same here, won't build with class name in definition.